### PR TITLE
K8SPSMDB-552: Disable SmartUpdate if cluster is unmanaged

### PR DESCRIPTION
**Pull request by egegunes**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| operator-self-healing | passed |
| smart-update | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| version-service | passed |
| operator-self-healing-chaos | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/759/commits/a11807788bb294616e32ae8b80d79cd9d656338b
image: `perconalab/percona-server-mongodb-operator:PR-759-a1180778`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| demand-backup-sharded | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| operator-self-healing | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| operator-self-healing-chaos | passed |
| pitr | passed |
| pitr-sharded | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/759/commits/0ad03f9fc19d09fa86520f95b4d7425abf238f08
image: `perconalab/percona-server-mongodb-operator:PR-759-0ad03f9f`

File: pkg/apis/psmdb/v1/psmdb_defaults.go [hunk 1]
```diff
@@ -438,6 +438,10 @@ func (cr *PerconaServerMongoDB) CheckNSetDefaults(platform version.Platform, log
 		return errors.New("backup.enabled must be false on unmanaged clusters")
 	}
 
+	if cr.Spec.Unmanaged && cr.Spec.UpdateStrategy == SmartUpdateStatefulSetStrategyType {
```
Comment (delgod) [inline pkg/apis/psmdb/v1/psmdb_defaults.go:441]: should we set `cr.Spec.ClusterServiceDNSSuffix` before return from the function?

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/apis/psmdb/v1/psmdb_defaults.go (hunk 1 above)
Comment (egegunes) [reply pkg/apis/psmdb/v1/psmdb_defaults.go]: fixed

**Review by nonemax — APPROVED** [review]

**Review by defbin — APPROVED** [review]
lgtm

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| limits | passed |
| demand-backup | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| operator-self-healing | passed |
| smart-update | passed |
| upgrade | passed |
| operator-self-healing-chaos | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/759/commits/594d3450e4953204bb004166dba8df98916c209a
image: `perconalab/percona-server-mongodb-operator:PR-759-594d3450`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| monitoring-2-0 | failed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| arbiter | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| operator-self-healing-chaos | passed |
| liveness | passed |
| smart-update | passed |
| pitr | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | passed |
| non-voting | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/759/commits/8a88e87f07eaff71d09e675a2b15ef6ba6f929e7
image: `perconalab/percona-server-mongodb-operator:PR-759-8a88e87f`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | passed |
| storage | passed |
| init-deploy | passed |
| self-healing | passed |
| limits | passed |
| self-healing-chaos | passed |
| scaling | passed |
| operator-self-healing | passed |
| security-context | passed |
| operator-self-healing-chaos | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/759/commits/8a88e87f07eaff71d09e675a2b15ef6ba6f929e7
image: `perconalab/percona-server-mongodb-operator:PR-759-8a88e87f`

**Pull request merged**